### PR TITLE
[xml] fix #1666 wrong cdata rule description and examples

### LINKED ISSUE #1666: [xml] wrong cdata rule description and examples
**Rule:** [MistypedCDATASection](https://pmd.github.io/pmd-6.11.0/pmd_rules_xml_errorprone.html#mistypedcdatasection)

**Description:**
The description and example of the cdata xml rule is wrong.
The start token is not `<!CDATA[`, but `<![CDATA[`

### PR DESCRIPTION
**Pull request by KroArtem**
Before submitting a PR, please check that:
 - [x] The PR is submitted against `master`. The PMD team will merge back to support branches as needed.
 - [x] `./mvnw clean verify` passes. This will [build](https://github.com/pmd/pmd/blob/master/BUILDING.md) and test PMD, execute PMD and checkstyle rules. [Check this for more info](https://github.com/pmd/pmd/blob/master/CONTRIBUTING.md#code-style)

**PR Description:**
Fixes #1666 

Fixed description. Example section needs clarifying, though

### CODE REVIEW

**Comment by pmd-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">No java rules are changed!</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: pmd-xml/src/main/resources/category/xml/errorprone.xml [hunk 1]
```diff
@@ -30,7 +30,7 @@ An XML CDATA section begins with a &lt;!CDATA[ marker, which has only one [, and
         </properties>
         <example>
 <![CDATA[
-An extra [ looks like &lt;!CDATA[[]]&gt;, and an extra ] looks like &lt;!CDATA[]]]&gt;.
+An extra [ looks like &lt;![CDATA[[]]&gt;, and an extra ] looks like &lt;![CDATA[]]]&gt;.
```
Comment (adangel) [inline pmd-xml/src/main/resources/category/xml/errorprone.xml]: See https://github.com/pmd/pmd/issues/1666#issuecomment-502348118 for suggestions for the examples.

Comment (adangel) [reply pmd-xml/src/main/resources/category/xml/errorprone.xml]: We could maybe also improve the description a bit...

Comment (adangel) [reply pmd-xml/src/main/resources/category/xml/errorprone.xml]: I'll finish this

Comment (KroArtem) [reply pmd-xml/src/main/resources/category/xml/errorprone.xml]: Sorry for losing momentum, was busy with other stuff. I've read your comments in the issue regarding cdata so they were not worthless. :)

**Pull request merged**